app/Http/Controllers/BeatmapsController.php

```diff
@@ -52,29 +53,55 @@ public function scores($id)
 
         $beatmap = Beatmap::findOrFail($id);
         $mode = Request::input('mode', Beatmap::modeStr($beatmap->playmode));
+        $enabled_mods = Request::input('enabledMods');
 
         try {
-            $scores = $beatmap
+            $query = $beatmap
                 ->scoresBest($mode)
                 ->defaultListing()
                 ->with('user');
         } catch (\InvalidArgumentException $ex) {
             return error_popup($ex->getMessage());
         }
 
+        if ($enabled_mods) {
+            if (!$user->isSupporter()) {
+                return error_popup(trans('errors.supporter_only'));
+            }
+
+            $query->whereIn('enabled_mods', [$enabled_mods, 0]);
```

Comment:
that doesn't sound right...?

Also I just realized in current form,

score1: mod1, mod2, mod3
score2: mod1
score3: -

filtering by mod1 would return score2 and score3 while it should be score1 and score2?